bench-vortex/src/lib.rs

```diff
@@ -31,11 +31,14 @@ use crate::medicare_data::medicare_data_csv;
 use crate::reader::BATCH_SIZE;
 use crate::taxi_data::taxi_data_parquet;
 
-mod data_downloads;
+pub mod data_downloads;
 pub mod medicare_data;
+pub mod public_bi_data;
 pub mod reader;
 pub mod taxi_data;
 
+/// Creates a file in the data directory if it doesn't already exist.
+/// NB: Takes a relative path (name) and creates a path in the data directory.
 pub fn idempotent<T, E>(name: &str, f: impl FnOnce(&Path) -> Result<T, E>) -> Result<PathBuf, E> {
```

Comment:
so this is `idempotent2(data_path(name), f)`? my intuition here is to have 

```
pub trait IdempotentPath {
    fn to_path(&self) -> Path
}
``` 

then do 
```
impl IdempotentPath for str {...}
```

and for Path

Reply:
The original idempotent implementation has some usage elsewhere that I just kind of wanted to leave alone, but I'm happy to rewrite it this way.

Reply:
just rewrite it, this is not api but internal